### PR TITLE
update(JS): web/javascript/reference/iteration_protocols

### PR DESCRIPTION
**Pull request by undead404**
Оригінальний вміст: [Протоколи ітерування@MDN](https://developer.mozilla.org/en-us/docs/Web/JavaScript/Reference/Iteration_protocols), [сирці Протоколи ітерування@GitHub](https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/iteration_protocols/index.md)

Нові зміни:
- [mdn/content@4c26e8a](https://github.com/mdn/content/commit/4c26e8a3fb50d06963b06017f51ce19364350564)
- [mdn/content@9cfdeb4](https://github.com/mdn/content/commit/9cfdeb4750a620b3a9e76b115b4552b8a9322d74)
- [mdn/content@6d60617](https://github.com/mdn/content/commit/6d606174faaedaa5dee7b7ebd87602cd51e5dd7e)
- [mdn/content@c227429](https://github.com/mdn/content/commit/c2274293475b0a5b4febf85a49c1f91bf43ebac7)
- [mdn/content@f29d8a6](https://github.com/mdn/content/commit/f29d8a648ef7ada05a00c358bfb9e9f67f3bc90d)
- [mdn/content@1443324](https://github.com/mdn/content/commit/144332479d3f90d12008866bc91df77da62451c2)
- [mdn/content@512c1d5](https://github.com/mdn/content/commit/512c1d5b115ea87d4b7943a9cd6b50b44ba09d0b)
- [mdn/content@ab97df6](https://github.com/mdn/content/commit/ab97df6ce8865569507bcfc884206a1ed297a690)
- [mdn/content@9c4fb23](https://github.com/mdn/content/commit/9c4fb236cd9ced12b1eb8e7696d8e6fcb8d8bad3)
- [mdn/content@5494a44](https://github.com/mdn/content/commit/5494a442f15c23f62c6374985e670e5ff99f1562)
- [mdn/content@3bf2463](https://github.com/mdn/content/commit/3bf2463d6d13f1f4b388ff2a0daaea9da5e545eb)

### CODE REVIEW

File: files/uk/web/javascript/reference/iteration_protocols/index.md [hunk 1]
```diff
@@ -101,6 +94,8 @@ console.log(aGeneratorObject[Symbol.iterator]() === aGeneratorObject);
 // true: метод @@iterator повертає сам об'єкт (сам ітератор), тож це ітерований ітератор
 ```
 
+Усі вбудовані ітератори успадковують від {{jsxref("Iterator", "Iterator.prototype")}}, котрий має реалізацію методу `[@@iterator]()`, що повертає `this`, тож вбудовані ітератори також є ітерованими об'єктами.
```
Comment (KlashReik) [inline files/uk/web/javascript/reference/iteration_protocols/index.md:97]:
```suggestion
Усі вбудовані ітератори успадковують від {{jsxref("Iterator", "Iterator.prototype")}}, котрий має реалізацію методу `[@@iterator]()`, що повертає `this`, тож вбудовані ітератори також є ітерованими об'єктами.

Не можу зрозуміти зміст. Успадковують від Iterator.prototype що? Може просто успадковують Iterator.prototype чи успадковуються від Iterator.prototype?
```

Comment (undead404) [reply files/uk/web/javascript/reference/iteration_protocols/index.md:97]: Успадковують усе. Він є в їх ланцюжку прототипів

Comment (KlashReik) [reply files/uk/web/javascript/reference/iteration_protocols/index.md:97]:
```suggestion
Усі вбудовані ітератори успадковуються від {{jsxref("Iterator", "Iterator.prototype")}}, котрий має реалізацію методу `[@@iterator]()`, що повертає `this`, тож вбудовані ітератори також є ітерованими об'єктами.
```

Тоді може так?

Comment (undead404) [reply files/uk/web/javascript/reference/iteration_protocols/index.md:97]:
```suggestion
Усі вбудовані ітератори мають в ланцюжку прототипів об'єкт {{jsxref("Iterator", "Iterator.prototype")}}, котрий має реалізацію методу `[@@iterator]()`, що повертає `this`, тож вбудовані ітератори також є ітерованими об'єктами.
```
Якщо дуже муляє, то краще так

**Pull request merged**